lib/std/special/compiler_rt/clzsi2.zig

```diff
@@ -0,0 +1,150 @@
+// Ported from:
+//
+// https://github.com/llvm-mirror/compiler-rt/blob/f0745e8476f069296a7c71accedd061dce4cdf79/lib/builtins/clzsi2.c
+// https://github.com/llvm-mirror/compiler-rt/blob/f0745e8476f069296a7c71accedd061dce4cdf79/lib/builtins/arm/clzsi2.S
+const builtin = @import("builtin");
+
+// Precondition: a != 0
+fn __clzsi2_generic(a: i32) callconv(.C) i32 {
+    @setRuntimeSafety(builtin.is_test);
+
+    var x: c_uint = @bitCast(c_uint, a);
```

Comment:
Please use the `iN`/`uN` types as it makes it easier to reason about their effective size.

Reply:
I tried to use them as first but it caused too many interger overflow, once I noted that __udivsi3 was using c_uint and c_int to avoid this problem and be closer to the C original source

Reply:
Don't sweep the overflow problems under the rug! Bitcast the `i32` argument to `u32` and use it.